src/app/components/Card.tsx

```diff
@@ -0,0 +1,89 @@
+import { CardType } from '../api/card';
+
+export interface Props {
+  cardType: number;
+  juz?: number;
+  surah?: number;
+  surahName?: string;
+  start?: number;
+  end?: number;
+  murojaahMethod: string;
+  totalMurojaah: number;
+  occuredAt: string;
+}
+
+const cardClassnames: Record<string, string> = {
+  container: 'p-4 mb-5 bg-teal text-white rounded-lg',
+  title: 'text-xl font-black',
+  data: 'font-normal',
+  date: 'font-extralight',
+};
+
+export function Card(props: Props): JSX.Element {
+  let el: JSX.Element = <></>;
+  switch (props.cardType) {
+    case CardType.Juz:
+      el = juzCard(props);
+      break;
+    case CardType.Surah:
+      el = surahCard(props);
+      break;
+    case CardType.Ayah:
+      el = ayahCard(props);
+      break;
+  }
+
+  return el;
+}
+
+function juzCard(props: Props): JSX.Element {
```

Comment:
if this is a function, lets rename it as verb like `renderJuzCard`. but if this is component, then lets use `CamelCase` and define in `const xxx = () => ()`